### DISCUSSION TITLE
错误扇区移除操作

### DISCUSSION THREAD

**Discussion by summerxue1992** [filecoin-project/community-china#80] [软件优化]
如何移除worker上工作到一半失败的扇区。lotus-miner sectors remove --really-do-itr  [id]后 ，miner上查看扇区信息，依然存在

**Comment by summerxue1992** [comment]
怎么样，才能把错误的扇区(历史原因，运行到一半p1阶段的扇区给移除掉)。

**Reply by BBQFIL** [reply]
这要看这些P1任务是否还参与调度，调度和不调度的处理方法不一样。

**Reply by summerxue1992** [reply]
怎么知道还是否在参与调度呢

**Reply by BBQFIL** [reply]
先在miner上用lotus-miner sectors status --log $sectorId  看下这个扇区现在是什么状态，如果status列的状态结尾是failed。那可以用lotus-miner sectors remove --really-do-it $sectorId  终止这个扇区。   
其他状态扇区是无法着这样终止的。首先需要他调度成功，显示在lotus-miner sealing jobs 列表里，通过lotus-miner sealing abort ID   让他变成failed，然后再lotus-miner sectors remove --really-do-it $sectorId 终止这个扇区

**Reply by BBQFIL** [reply]
上边操作完后，实际上还要重启一下worker进程，这个扇区才会真正的结束。

**Reply by aizhikai** [reply]
lotus-miner sealing jobs 里面没有呢？ 怎么终止

**Reply by maxvint** [reply]
@summerxue1992 https://github.com/filguard/lotus-ops/blob/master/documents/questions.md#1-%E9%A1%BD%E5%9B%BA%E6%89%87%E5%8C%BA%E5%A6%82%E4%BD%95%E5%88%A0%E9%99%A4 看一下这篇，如何删除顽固扇区。
